### ISSUE TITLE
Pivotal and Create react app

### ISSUE THREAD

**Issue by Lube** [pivotal-cf/pivotal-ui#487]
Hey guys Im trying to test some components on a project made with create-react-app and im getting this error after installing notificactions package.

```
Failed to compile.

Error in ./~/pui-react-iconography/iconography.js
Module not found: 'babel-loader' in /home/....../node_modules/pui-react-iconography

 @ ./~/pui-react-iconography/iconography.js 62:13-96
```

I installed babel loader, but I suspect it probably is because of react-scripts default webpack conf.

Anyone successfully integrated cra with pivotal components?

**Comment by Lube** [comment]
I solved my issue by ejecting the webpack conf and adding the babel-loader to the pui react iconography module.

```
// Process JS with Babel.
      {
        test: /\.(js|jsx)$/,
        include: [paths.appSrc, paths.nodePaths + '/pui-react-iconography'],
        loader: 'babel',
        query: {
          
          // This is a feature of `babel-loader` for webpack (not Babel itself).
          // It enables caching results in ./node_modules/.cache/babel-loader/
          // directory for faster rebuilds.
          cacheDirectory: true
        }
      },
```

Is there a reason why iconogaphy needs babel-loader?

**Comment by charleshansen** [comment]
Iconography needs to set its own loaders to load svgs, specifically, it needs `svg-react-loader`. We found an issue in the past where create react app was adding a bunch of extra loaders that would break the svgs. To fix this, we used `!!babel-loader!svg-react-loader!` in our `require` statement. The `!!` at the front tells it to ignore everything else from the webpack config. The `babel-loader` is used because we still need to convert the jsx somehow. At the time, it did work with CRA.

I just tried it myself and got the same error. I'm guessing the CRA scripts changed in some important way since last time we made it work with them. Not quite sure how to fix this right now...

**Comment by Lube** [comment]
I don't have a strong grasp on webpack and the loader ecosystem, but maybe this is useful for someone who does.

Btw im using pui-react-svg and according to the docs it assumes that you have svgs in app/svg in your root folder, and throws a warning if it doesn't exist, any way to supress this warning?

Thanks for this effort btw the components are super slick and the docs site rocks.

**Comment by charleshansen** [comment]
- It is a legit bug, thanks for bringing it up.

- I've tried to suppress the svg warning. Can't do it. Webpack really needs to check all possible directories where things might live, even if those directories might not exist.

- Welcome. Glad you are enjoying it.

**Comment by duncanfinney** [comment]
+1 for this to be fixed. Loving PivotalUI, but this bug is a bit annoying. 

create-react-app supports requiring SVG files out of the box. Just removing `!babel-loader!!react-svg-loader!` from the require statement makes it compile for me, but it doesn't work just like that because of the way the src property is set.

Maybe we could set/check an environment variable.

SVG component: 
```
render()  {
    if (process.env.REACT_APP_PUI) {
        const component = require('../.../app/svgfile.svg');
        return <component />
    }

    //...Otherwise do it like it is now
}
```

**Comment by dovibergman** [comment]
+1 for please do solve. I am getting build problems because of this issue in my application, which use the pui-react-table and by extension iconography as well

**Comment by charleshansen** [comment]
I think I finally have a solution that both compiles and renders the svgs correctly (there are ways to make it compile, but the svgs will either be missing or completely unstyled). Assuming it works all the way, at least the important part should be published in 8.2.0. I'm hoping within a week or two.

For what it is worth, the fix is simply to add the relevant loaders as dependencies to pui-react-iconography and pui-react-svg and then also make sure they know to use the react babel-preset. This didn't work for me when I tried it much earlier, but looks like it works now.

**Comment by charleshansen** [comment]
Just pushed the commit 486edaecf6f2c56800a2641755a3a4fc57907849

**Comment by charleshansen** [comment]
published in 8.3.0
